src/h5web/toolbar/controls/DomainSlider/BoundEditor.tsx

```diff
@@ -14,14 +20,27 @@ interface Props {
   onChange: (val: number) => void;
 }
 
-function BoundEditor(props: Props) {
+interface Handle {
+  cancel: () => void;
+}
+
+const BoundEditor = forwardRef<Handle, Props>((props, ref) => {
   const { bound, value, isEditing, hasError, onEditToggle, onChange } = props;
 
   const id = `${bound}-bound`;
   const inputRef = useRef<HTMLInputElement>(null);
 
   const [inputValue, setInputValue] = useState('');
 
+  function cancel() {
+    onEditToggle(false);
+    setInputValue(formatPreciseValue(value));
+  }
+
+  /* Expose `cancel` function to parent component through ref handle so that
+    `inputValue` can be reset when the user closes the domain tooltip. */
+  useImperativeHandle(ref, () => ({ cancel }));
```

Comment:
First time I use this hook. The reason is as follows:

When the user closes the domain tooltip with Escape or by clicking outside of it, we want to cancel any edits made to the min/max fields. This means turning off edit mode (which is done in `DomainSlider`, but also resetting the `inputValue` here, in `BoundEditor`. Since the slider and tooltip components don't have access to the `inputValue` state, they can't reset it themselves. `useImperativeHandle` allows exposing the `cancel` function to the parent component via a forwarded `ref`: the parent component can then set a `ref` on each bound editor and call `ref.current.cancel()`.